### DISCUSSION TITLE
WA Business Tags

### DISCUSSION THREAD

**Discussion by graficar** [devlikeapro/waha#427] [General]
Is it possible to tag WA Business contacts with the API?
Thanks for your help.

[![patron:PLUS](https://img.shields.io/badge/patron-PLUS-a0e6ba)](https://waha.devlike.pro/docs/how-to/plus-version/#tiers)

**Comment by devlikepro** [comment]
Hi! tags here means labels, right? If yes - we have such request, hopefully will handle soon :crossed_fingers: 
https://github.com/devlikeapro/waha/issues/318

[![patron:PRO](https://img.shields.io/badge/patron-PRO-188a42)](https://waha.devlike.pro/docs/how-to/plus-version/#tiers)

**Comment by graficar** [comment]
When do you plan to implement that functionality?

[![patron:PLUS](https://img.shields.io/badge/patron-PLUS-a0e6ba)](https://waha.devlike.pro/docs/how-to/plus-version/#tiers)

**Reply by devlikepro** [reply]
this quarter (July-September) :crossed_fingers: 

[![patron:PRO](https://img.shields.io/badge/patron-PRO-188a42)](https://waha.devlike.pro/docs/how-to/plus-version/#tiers)

**Comment by devlikepro** [comment]
working on it, will be in `2024.8.1` version!

[![patron:PRO](https://img.shields.io/badge/patron-PRO-188a42)](https://waha.devlike.pro/docs/how-to/plus-version/#tiers)

**Comment by devlikepro** [comment]
You can work with [WhatsApp Labels](https://faq.whatsapp.com/3398508707096369/?cms_platform=android) available in **WhatsApp Business** using the API!

## Features

Here's the list of features that are available by [**🏭 Engines**](https://waha.devlike.pro/docs/how-to/engines/):

|                                                                          | WEBJS | NOWEB |
|--------------------------------------------------------------------------|-------|-------|
| **Get list of labels**<br>`GET /api/{session}/labels`                    | ✔️    | ✔️    |
| **Get chats by label id**<br>`GET /api/{session}/labels/{labelId}/chats` | ✔️    | ✔️    |
| **Get labels by chat id**<br>`GET /api/{session}/labels/chats/{chatId}/` | ✔️    | ✔️    |
| **PUT labels to chat**<br>`PUT /api/{session}/labels/chats/{chatId}/`    | ✔️    | ✔️    |
| `labels.upsert`                                                          |       | ✔️    |
| `labels.deleted`                                                         |       | ✔️    |
| `labels.chat.added`                                                      |       | ✔️    |
| `labels.chat.deleted`                                                    |       | ✔️    |

****

_If you see the feature is not available in the above list, please [create a feature request](https://github.com/devlikeapro/waha/issues/new/choose) or **leave "+1" comment** on the existing one._
****


## Endpoints

### Get list of labels

You can get a list of labels for the session using the endpoint:

```bash
GET /api/{session}/labels
```

Response:

```json

[
  {
    "id": "1",
    "name": "New Client",
    "color": 1,
    "colorHex": "#64c4ff"
  },
  ...
]
```

### Get chats by label id

You can get a list of chats by label id using the endpoint:

```bash
GET /api/{session}/labels/{labelId}/chats
```

ℹ️ Response right now depends on
[**🏭 Engine**](https://waha.devlike.pro/docs/how-to/engines/)
you're using,
the same way as
[**💬 Chats**](https://waha.devlike.pro/docs/how-to/chats/#get-all-chats)

### Get labels by chat id

You can get a list of labels by chat id using the endpoint:

```bash
GET /api/{session}/labels/chats/{chatId}/
```

Response:

```json
[
  {
    "id": "1",
    "name": "New Client",
    "color": 1,
    "colorHex": "#64c4ff"
  },
  ...
]
```

### Update labels to chat

```bash
PUT /api/{session}/labels/chats/{chatId}/
```

**Upsert label**:

👉 You need to provide **the full list of labels** you want to set to the chat. All other labels will be removed.

```json
{
  "labels": [
    {
      "id": "1"
    }
  ]
}
```

**Remove labels**:

```json
{
  "labels": []
}

```

## Webhooks

### labels.upsert

```json
{
  "event": "label.upsert",
  "session": "default",
  "payload": {
    "id": "10",
    "name": "Label Name",
    "color": 14,
    "colorHex": "#00a0f2"
  },
  "engine": "NOWEB",
  ...
}

```

### labels.deleted

```json
{
  "event": "label.deleted",
  "session": "default",
  "payload": {
    "id": "10",
    "name": "",
    "color": 14,
    "colorHex": "#00a0f2"
  },
  "engine": "NOWEB",
  ...
}

```

### labels.chat.added

```json
{
  "event": "label.chat.added",
  "session": "default",
  "payload": {
    "labelId": "6",
    "chatId": "11111111111@c.us",
    "label": null <=== right after scanning QR it can be null. 
  },
  "engine": "NOWEB",
  ...
}
```

### labels.chat.deleted

```json
{
  "event": "label.chat.deleted",
  "session": "default",
  "payload": {
    "labelId": "6",
    "chatId": "11111111111@c.us",
    "label": null
  },
  "engine": "NOWEB",
  ...
}
```


[![patron:PRO](https://img.shields.io/badge/patron-PRO-188a42)](https://waha.devlike.pro/docs/how-to/plus-version/#tiers)
